helm/cloudnative-pg/templates/cilium-network-policy.yaml

```diff
@@ -41,4 +42,74 @@ spec:
       toPorts:
       - ports:
         - port: "8080"
+---
+# This policy targets the Postgres database instance pods managed by CNPG.
+apiVersion: cilium.io/v2
+kind: CiliumClusterwideNetworkPolicy
+metadata:
+  name: {{ include "cloudnative-pg.fullname" . }}-db-instance
+spec:
+  endpointSelector:
+    matchLabels:
+      cnpg.io/podRole: instance
+  ingress:
+    # Allow ingress from CNPG operator.
+    - fromEndpoints:
+        - matchLabels:
+            app.kubernetes.io/instance: cloudnative-pg
+            app.kubernetes.io/name: cloudnative-pg
+          matchExpressions:
+            - key: io.kubernetes.pod.namespace
+              operator: Exists
+      toPorts:
+        - ports:
+            - port: "8000"
+        - ports:
+            - port: "5432"
+    # Allow Prometheus scraping.
+    - fromEntities:
+        - cluster
+      toPorts:
+        - ports:
+            - port: "9187"
+  egress:
+    - toEntities:
+        - kube-apiserver
```

Comment:
Why does this pod need api server? Is this one the actual operator?

Reply:
There is a management process that sits next to postgres and updates cluster status in the API